### PR TITLE
chore: cleanup pom files

### PR DESCRIPTION
**Pull request by mjsax**
### Description 
_What behavior do you want to change, why, how does your patch achieve the changes?_

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-engine/pom.xml [hunk 1]
```diff
@@ -197,23 +197,5 @@
                 </configuration>
             </plugin>
         </plugins>
-
```
Comment (mjsax) [inline ksqldb-engine/pom.xml:200]: This and the other 3 modules don't have any resource folder and don't need to get the version. Thus, it seems appropriate to just delete those definitions in the pom (might have be added via c&p?)

**Comment by guozhangwang** [comment]
What this PR is for?

**Comment by mjsax** [comment]
We saw some test failures (I could reproduce locally at some point but not any longer), for which `CliTest` failed because it could not get the version from `AppInfo` -- instead of providing the version string (ie, `"7.0.0-0"`) or similar, the variable in `version.properties` was not replaced and `AppInfo` returned string `"${project.version}"`.

It all still somewhat a mystery...

File: ksqldb-common/pom.xml [hunk 2]
```diff
@@ -144,14 +144,6 @@
         </plugins>
 
         <resources>
-            <resource>
-                <directory>src/main/resources</directory>
-                <filtering>true</filtering>
-                <includes>
-                    <include>ksql-version.streamsProperties</include>
```
Comment (mjsax) [inline ksqldb-common/pom.xml:151]: This file does not exist any longer

**Review by agavra — APPROVED** [review]

**Pull request merged**